### DISCUSSION TITLE
how to bind to properties rather then attributes?

### DISCUSSION THREAD

**Discussion by jimmywarting** [alpinejs/alpine#2886] [1. Help]
for things that isn't defined with attributes such as indeterminate on checkboxes... how can i set it?

tried things like:
```html
<input type="checkbox" :indeterminate="boolean">
```
but it only seems to do things with attributes?

**Comment by browner12** [comment]
AFAIK `:indeterminate` is a CSS pseudo class that helps you style the element, unrelated to Alpine.

so if you're using something like Tailwind, you can do

```html
<input type="checkbox" class="appearance-none indeterminate:bg-gray-300 ..." />
```

**Reply by jimmywarting** [reply]
indeterminate is also a IDL property (not just a css stuff)
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox#indeterminate_state_checkboxes

**Reply by browner12** [reply]
Gotcha.

I think Alpine can only bind to content attributes, not IDL attributes. So I think you'd need use javascript triggered by an event.

```html
<input type="checkbox" x-on:click="$el.indeterminate = true" />
```

**Reply by jimmywarting** [reply]
I tough so, would be nice to be able to bind to element properties too that don't have things as attributes.
